infra/Pos/Communication/Relay/Logic.hs

```diff
@@ -338,27 +348,50 @@ relayWorkersImpl
        , MonadReportingMem m
        , Message Void
        )
-    => OutSpecs -> ([WorkerSpec m], OutSpecs)
-relayWorkersImpl allOutSpecs =
+    => RelayContext m -> OutSpecs -> ([WorkerSpec m], OutSpecs)
+relayWorkersImpl relayContext allOutSpecs =
     first (:[]) $ worker allOutSpecs $ \sendActions ->
         handleAll handleWE $ reportingFatal version $ action sendActions
   where
+    action :: SendActions m -> m ()
     action sendActions = do
-        queue <- _rlyPropagationQueue <$> askRelayMem
+        let sendIt :: PropagationMsg -> Maybe NodeId -> NodeId -> Microsecond -> m ()
+            sendIt = \message _ peer timeInQueue -> do
+                jsonLog $ EnqueueDequeueTime $ fromIntegral $ timeInQueue
+                case message of
+                    InvReqDataPM key contents -> do
+                        logDebug $ sformat
+                            ("Propagation data with key: "%build) key
+                        withConnectionTo sendActions peer $ \_ ->
+                            pure $ Conversation $ irdHandler key contents
+                    DataOnlyPM contents -> do
+                        logDebug $ sformat
+                            ("Propagation data: "%build) contents
+                        withConnectionTo sendActions peer $ \_ ->
+                            pure $ Conversation $ doHandler contents
+        forever $ do
+            let dequeue = _rlyDequeue relayContext
+            (msg, provenance, peer, time) <- dequeue
+            sendIt msg provenance peer time
```

Comment:
So currently the simple relayer from TW only provides the enqueue and dequeue, and CSL still needs to provide the relayer thread itself. But in our next iteration, TW will also provide that relayer thread, and so this bit here will change again.